lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -3299,6 +3301,40 @@ instance IsServerError ErrSignPayment where
         ErrSignPaymentWithRootKey e@ErrWithRootKeyWrongPassphrase{} -> toServerError e
         ErrSignPaymentIncorrectTTL e -> toServerError e
 
+instance IsServerError ErrWitnessTx where
+    toServerError = \case
+        ErrWitnessTxSignTx e -> toServerError e
+        ErrWitnessTxNoSuchWallet e -> (toServerError e)
+            { errHTTPCode = 404
+            , errReasonPhrase = errReasonPhrase err404
+            }
+        ErrWitnessTxWithRootKey e@ErrWithRootKeyNoRootKey{} -> (toServerError e)
+            { errHTTPCode = 403
+            , errReasonPhrase = errReasonPhrase err403
+            }
+        ErrWitnessTxWithRootKey e@ErrWithRootKeyWrongPassphrase{} -> toServerError e
+        ErrWitnessTxIncorrectTTL e -> toServerError e
+
+instance IsServerError ErrSignTx where
+    toServerError = \case
+        ErrSignTxKeyNotFoundForAddress addr ->
+            apiError err500 KeyNotFoundForAddress $ mconcat
+                [ "That's embarrassing. I couldn't sign the given transaction: "
+                , "I haven't found the corresponding private key for a known "
+                , "input address I should keep track of: ", showT addr, ". "
+                , "Retrying may work, but something really went wrong..."
+                ]
+        ErrSignTxInvalidSerializedTx hint ->
+            apiError err500 CreatedInvalidTransaction hint
+        ErrSignTxInvalidEra ->
+            apiError err500 CreatedInvalidTransaction $ mconcat
+                [ "Whoops, it seems like I just experienced a hard-fork in the "
+                , "middle of other tasks. This is a pretty rare situation but "
+                , "as a result, I must throw-away what I was doing. Please "
+                , "retry whatever you were doing in a short delay."
```

Comment:
```suggestion
                , "as a result, I must throw away what I was doing. Please "
                , "retry your request."
```

Hi @paweljakubas: I'm just wondering, how long should the user wait here? Is it a case of a few seconds, or a few minutes?